packages/token/src/common/erc2981/erc2981.cairo

```diff
@@ -0,0 +1,221 @@
+        default_royalty_info: RoyaltyInfo,
+        token_royalty_info: Map<u256, RoyaltyInfo>,
+    }
+
+    mod Errors {
+        pub const INVALID_ROYALTY: felt252 = 'ERC2981: invalid royalty';
+        pub const INVALID_ROYALTY_RECEIVER: felt252 = 'ERC2981: invalid receiver';
+    }
+
+    /// Constants expected to be defined at the contract level used to configure the component
+    /// behaviour.
+    ///
+    /// - `FEE_DENOMINATOR`: The denominator with which to interpret the fee set in
+    ///   `set_token_royalty` and `set_default_royalty` as a fraction of the sale price.
+    pub trait ImmutableConfig {
+        const FEE_DENOMINATOR: u128;
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(ERC2981Impl)]
+    impl ERC2981<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl Immutable: ImmutableConfig,
+        impl SRC5: SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>,
+    > of IERC2981<ComponentState<TContractState>> {
+        /// Returns how much royalty is owed and to whom, based on a sale price that may be
+        /// denominated in any unit of exchange. The royalty amount is denominated and should be
+        /// paid in that same unit of exchange.
+        ///
+        /// The returned tuple contains:
+        ///
+        /// - `t.0`: The receiver of the royalty payment.
+        /// - `t.1`: The amount of royalty payment.
+        fn royalty_info(
+            self: @ComponentState<TContractState>, token_id: u256, sale_price: u256
+        ) -> (ContractAddress, u256) {
+            let token_royalty_info = self.token_royalty_info.read(token_id);
+
+            // If the token has no specific royalty info, use the default.
+            let royalty_info = if token_royalty_info.receiver.is_zero() {
+                self.default_royalty_info.read()
+            } else {
+                token_royalty_info
+            };
+
+            let royalty_amount = sale_price
+                * royalty_info.royalty_fraction.into()
+                / Immutable::FEE_DENOMINATOR.into();
+
+            (royalty_info.receiver, royalty_amount)
+        }
+    }
+
+    //
+    // Internal
+    //
+
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl Immutable: ImmutableConfig,
+        impl SRC5: SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of InternalTrait<TContractState> {
+        /// Initializes the contract by setting the default royalty.
+        fn initializer(
+            ref self: ComponentState<TContractState>,
+            default_receiver: ContractAddress,
+            default_royalty_fraction: u128
+        ) {
+            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
+            src5_component.register_interface(IERC2981_ID);
+
+            self.set_default_royalty(default_receiver, default_royalty_fraction)
+        }
```

Comment:
To my mind, it makes sense to add config validation to the initializer by checking the `FEE_DENOMINATOR` to be non-zero

Reply:
Agree, I will update.